cve_bin_tool/parsers/python.py

```diff
@@ -33,8 +33,8 @@ def run_checker(self, filename):
             )
         except subprocess.CalledProcessError as e:
             self.logger.error(e.output)
-            pip_version = subprocess.check_output(["pip3", "--version"])
-            pip_version = float(pip_version[6:10])
+            pip_version = str(subprocess.check_output(["pip3", "--version"]))
+            pip_version = float(".".join(pip_version.split(" ")[1].split(".")[:2]))
```

Comment:
Okay, I can see where we're going with this -- you're trying to truncate the version number to keep the float compare on the next line.  But I think we either need a comment explaining why it's so weird or just use `packaging.version` to do a version compare the way we do in the scanner.  Here's a snippet showing what using `packaging.version` would look like:

```python
from packaging.version import parse as parse_version

pip_version = "pip 20.0.2 from /usr/lib/python3/dist-packages/pip (python 3.8)"
pip_version = pip_version.split(" ")[1]
if parse_version(pip_version) < parse_version("22.2") :
    print("pip is too old")
else:
    print("all good")
```

Reply:
I've got a mild preference for using parse_version because I find it conceptually easier to compare the versions correctly than to truncate, but a good explanation and a float compare is maybe a tiny bit faster than loading the library if you prefer.